internal/command/arguments/plan_test.go

```diff
@@ -170,6 +179,94 @@ func TestParsePlan_targets(t *testing.T) {
 	}
 }
 
+func TestParsePlan_targetFile(t *testing.T) {
+	foobarbaz, _ := addrs.ParseTargetStr("foo_bar.baz")
+	boop, _ := addrs.ParseTargetStr("module.boop")
+	testCases := map[string]struct {
+		fileContent string
+		want        []addrs.Targetable
+		wantDiags   tfdiags.Diagnostics
+	}{
+		"target file valid single target": {
+			fileContent: "foo_bar.baz",
+			want:        []addrs.Targetable{foobarbaz.Subject},
+		},
+		"target file valid multiple targets": {
+			fileContent: "foo_bar.baz\nmodule.boop",
+			want:        []addrs.Targetable{foobarbaz.Subject, boop.Subject},
+		},
+		"target file invalid target": {
+			fileContent: "foo.",
+			want:        nil,
+			// wantDiags:   "Invalid target \"foo.\": Dot must be followed by attribute name",
+		},
+		"target file valid comment": {
+			fileContent: "#foo_bar.baz",
+			want:        nil,
+			// From spec:After trimming spaces, if the line starts with our typical comment
+			// character # or is an empty string then the line is completely ignored and
+			// parsing continues with the next line. (The other comment variants of //
+			// and /* ... */ would not be supported here, to keep this new format relatively simple.)
+		},
+		"target file valid spaces": {
+			fileContent: "   foo_bar.baz",
+			want:        []addrs.Targetable{foobarbaz.Subject},
+			// From spec: Each line is subjected to bytes.TrimSpace (or equivalent)
+			// before attempting to parse it. Along with removing leading and
+			// trailing spaces/tabs, this should also remove any trailing carriage return
+			// character that might be included if the file was written on a Windows
+			// system using the typical Windows line-ending convention.
+		},
+		"target file invalid bracket with spaces": {
+			fileContent: `    [boop]`,
+			want:        nil,
+			wantDiags: tfdiags.Diagnostics(nil).Append(
+				&hcl.Diagnostic{
+					Severity: hcl.DiagError,
+					Summary:  "Invalid syntax",
+					Detail:   `For target "    [boop]": Must begin with a variable name.`,
+					Subject: &hcl.Range{
+						// Filename set once tempFile created
+						Start: hcl.Pos{Line: 1, Column: 1},
+						End:   hcl.Pos{Line: 1, Column: 11, Byte: 10},
+					},
+				},
```

Comment:
Hey @apparentlymart – Would you provide a suggestion for what this _should_ look like? Once I understand that, then I'll rewrite the logic the business logic to accomplish the desired error.